eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -93,10 +93,22 @@ class MultiPartHandler(nodeParams: NodeParams, register: ActorRef, db: IncomingP
                   pendingPayments = pendingPayments + (p.add.paymentHash -> (record.paymentPreimage, handler))
               }
           }
-          case None =>
-            Metrics.PaymentFailed.withTag(Tags.Direction, Tags.Directions.Received).withTag(Tags.Failure, "InvoiceNotFound").increment()
-            val cmdFail = CMD_FAIL_HTLC(p.add.id, Right(IncorrectOrUnknownPaymentDetails(p.payload.totalAmount, nodeParams.currentBlockHeight)), commit = true)
-            PendingRelayDb.safeSend(register, nodeParams.db.pendingRelay, p.add.channelId, cmdFail)
+          case None => p.payload.keySend match {
+            case Some(keySend) =>
```

Comment:
You also need to check that key-send is active. If the user doesn't activate keySend in his features, he wants to reject those payments:

```suggestion
            case Some(keySend) if nodeParams.features.hasFeature(Features.KeySend) =>
```